ceph_deploy/util/pkg_managers.py

```diff
@@ -177,3 +177,93 @@ def zypper_refresh(conn):
         conn,
         cmd
     )
+
+
+class PackageManager(object):
+    """
+    Base class for all Package Managers
+    """
+
+    def __init__(self, remote_conn):
+        self.remote_info = remote_conn
+        self.remote_conn = remote_conn.conn
+
+    def _run(self, cmd):
+        return remoto.process.run(
+            self.remote_conn,
+            cmd
+        )
+
+class RPMManagerBase(PackageManager):
+    """
+    Base class to hold common pieces of Yum and DNF
+    """
+
+    def __init__(self, remote_conn, cmd):
+        super(RPMManagerBase, self).__init__(remote_conn)
+        self.cmd = cmd
+
+    def install(self, packages):
+        if isinstance(packages, str):
+            packages = [packages]
+
+        cmd = [
+            self.cmd,
+            '-y',
+            'install',
+        ]
+        cmd.extend(packages)
+        return self._run(cmd)
+
+    def remove(self, packages):
+        if isinstance(packages, str):
+            packages = [packages]
+
+        cmd = [
+            self.cmd,
+            '-y',
+            '-q',
+            'remove',
+        ]
+        cmd.extend(packages)
+        return self._run(cmd)
+
+    def clean(self, item=None):
+        item = item or 'all'
+        cmd = [
+            self.cmd,
+            'clean',
+            item,
+        ]
+
+        return self._run(cmd)
+
+
+class DNF(RPMManagerBase):
+    """
+    The DNF Package manager
+    """
+
+    def __init__(self, remote_conn):
+        super(DNF, self).__init__(remote_conn, 'dnf')
```

Comment:
@alfredodeza 
It would eliminate the need for `super`, but I don't quite see how it would work if `self.cmd` does not get set in `RPMManagerBase`.

If I do `Yum().install('tree')`, The `Yum` class does not override the `install` method -- it falls through to `RPMManagerBase`.    If `Yum` does not pass the executable name to `RPMManagerBase` via `__init__()`, how will it know which executable to use?

I don't think a class attribute would do that for me.

On a side note, I would be happy to rename `RPMManagerBase.cmd` to `executable` so it does not get confused with the `cmd` string I pass to `_run()`.